process/process.go

```diff
@@ -61,34 +61,102 @@ func (ps *processService) ProcessCreate(
 }
 
 type process struct {
-	pid      uint32                  // process id
-	root     string                  // root dir
-	cwd      string                  // current working dir
-	uid      uint32                  // effective uid
-	gid      uint32                  // effective gid
-	sgid     []int                   // supplementary groups
-	cap      cap.Capabilities        // process capabilities
-	status   map[string]string       // process status fields
-	nsInodes map[string]domain.Inode // process namespace inodes
-	ps       *processService         // pointer to parent processService
+	pid         uint32                  // process id
+	root        string                  // root dir
+	hroot       string                  // root dir from host perspective
+	cwd         string                  // current working dir
+	hcwd        string                  // current working dir from host perspective
+	uid         uint32                  // effective uid
+	gid         uint32                  // effective gid
+	sgid        []int                   // supplementary groups
+	cap         cap.Capabilities        // process capabilities
+	status      map[string]string       // process status fields
+	nsInodes    map[string]domain.Inode // process namespace inodes
+	initialized bool                    // process initialization completed
+	ps          *processService         // pointer to parent processService
 }
 
 func (p *process) Pid() uint32 {
+
+	if !p.initialized {
+		p.getInfo()
+	}
+
 	return p.pid
 }
 
 func (p *process) Uid() uint32 {
+
+	if !p.initialized {
+		p.getInfo()
+	}
+
 	return p.uid
 }
 
 func (p *process) Gid() uint32 {
+
+	if !p.initialized {
+		p.getInfo()
+	}
+
 	return p.gid
 }
 
-func (p *process) IsAdminCapabilitySet() bool {
+func (p *process) Cwd() string {
+
+	if !p.initialized {
+		p.getInfo()
+	}
+
+	return p.cwd
+}
+
+func (p *process) Root() string {
+
+	if !p.initialized {
+		p.getInfo()
+	}
+
+	return p.root
+}
+
+func (p *process) IsSysAdminCapabilitySet() bool {
```

Comment:
The IsXCapabilitySet() wrappers don't add much value in my view. The code is equally readable using p.isCapabilitySet(cap.EFFECTIVE, cap.X). Plus there are many caps, and we are not going to have wrappers for all.

Reply:
Reason for these wrappers is to avoid introducing an external dependency ("cap" type, which seats in sysbox-libs/capabilities/capability.go) into sysbox-fs' domain layer. This layer should be kept clean of external components (other than go stdlib pkgs).